lib/ring_logger/persistence.ex

```diff
@@ -0,0 +1,24 @@
+defmodule RingLogger.Persistence do
+  @moduledoc false
+
+  @spec load(String.t()) :: [RingLogger.entry()] | {:error, atom()}
+  def load(path) do
+    path
+    |> File.read!()
+    |> :erlang.binary_to_term()
+  rescue
+    error in File.Error ->
+      {:error, error.reason}
+
+    ArgumentError ->
+      {:error, :corrupted}
+  end
+
+  @spec save(String.t(), [RingLogger.entry()]) :: :ok | {:error, atom()}
+  def save(path, logs) do
+    File.write!(path, :erlang.term_to_binary(logs))
```

Comment:
[`term_to_binary/2` supports compression](https://www.erlang.org/doc/man/erlang.html#term_to_binary-2) if we can about having that in to potentially save some byte writes or space?

Reply:
> So is the goal here that we write the same logs + any new every 300 seconds (or whatever time configured?)
> 
> Would we want to instead have more of a rolling buffer? Like set a max buffer size, write to disk when we reach that limit? Though the downside here is losing what is still accumulating in the buffer in the event of a restart 🤔

Since RingLogger logs have to fit entirely in DRAM, they'll be small. I don't expect that we'd see any performance difference by trying to do things incrementally unless they're being saved to a slow NOR flash.

Regarding the "write the same logs + new ones", this is only storing the current state of RingLogger. Logs that have rotated out are still gone forever. This addresses the situation where you lose your logs when the RingLogger GenServer holding them restarts (due to a reboot or anything else). I think that if you want to store logs on disk in a way that's larger than what can be stored in memory, you should use a proper file-based logger backend so that you can get all of the file rotation, incremental appending, text-based viewing, etc.